src/main/java/com/example/animalshelter/controller/AnimalCreatedResponse.java

```diff
@@ -0,0 +1,12 @@
+package com.example.animalshelter.controller;
+
+import com.example.animalshelter.model.Animal;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@Data
+@AllArgsConstructor
+class AnimalCreatedResponse {
+
+    private final Animal animal;
```

Comment:
Normally we are not creating requests/responses to wrap domain objects.
This is problematic, especially when the domain object grows but endpoint responsibility stays the same.
This response should either contain only the id of created animal or a minimal amount of necessary fields for the client who will be using this endpoint.

Reply:
Thank you for explaining this! I have changed this and now endpoint return dto object with only id.